webdriver/tests/sessions/get_timeouts.py

```diff
@@ -0,0 +1,19 @@
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+
+def get_timeouts(session):
+    return session.transport.send("GET", "session/{session_id}/timeouts"
+                                  .format(session_id=session.session_id))
+
+
+# 8.4 Get Timeouts
+
+def test_get_timeouts(session):
+    # 8.4 step 1
+    response = get_timeouts(session)
+
+    assert isinstance(response.body["value"]["script"], int)
+    assert isinstance(response.body["value"]["implicit"], int)
+    assert isinstance(response.body["value"]["pageLoad"], int)
+    return assert_success(response)
```

Comment:
Why are you returning here?  Also, the response code ought be checked first. If it’s not 200, there’s no point in running the rest of these checks.

Reply:
I am returning assert_success because Step 8.4.2 says "Return success with data body." Is that the correct way to do it? Should I even return the data body at all?

Reply:
A test isn’t a literal implementation of the remote end steps: its purpose is, given some input, to check both the formal aspects and the semantics of a given command.

In this case the remote end is told to return a successful response and the test needs to assert that the implementation conforms to this behaviour. If however the response from `get_timeouts` is an error, the following line will error because there is no `value` field on `response.body`:

    assert isinstance(response.body["value"]["script"], int)

You need to first check whether the response was successful because this is a precursor for testing that the successful response’s JSON body is set up in the way it’s meant to be. In this case you need to change to test to do this:

    response = get_timeout(session)

    assert response.status == 200
    assert "value" in response.body
    assert isinstance(response.body["value"], dict)

    value = response.body["value"]
    assert "script" in value
    assert "implicit" in value
    assert "pageLoad" in value
    assert isinstance(value["script"], int)
    assert isinstance(value["implicit"], int)
    assert isinstance(value["pageLoad"], int)

Reply:
Your last update almost fixes the issues I mentioned in the last comment. However, there is no need for this:

    return assert_success(response)

You are correct that the remote end is being asked to return a successful response. The test isn’t.

Reply:
fixed